src/Events/Models/CloudEvent.cs

```diff
@@ -90,5 +90,25 @@ public static CloudEvent Deserialize(string jsonString)
         {
             return JsonSerializer.Deserialize<CloudEvent>(jsonString, new JsonSerializerOptions { });
         }
+
+        /// <summary>
+        /// Validated required properties Id, Time, Source, SpecVersion, Type and Subject
+        /// </summary>
+        /// <returns>A boolean indicating whether all required fields have a value</returns>
+        public bool ValidateRequiredProperties()
+        {
+            if (string.IsNullOrEmpty(Id) ||
+                Time == null ||
```

Comment:
Looks like `time` should be optional. I'm not sure how we can ensure conformence with RFC 3339 without making use of the cloudevents C# SDK.
https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/spec.md#time